# Table of Contents
 <p><div class="lev1"><a href="#Popularity-of-Music-Records"><span class="toc-item-num">1&nbsp;&nbsp;</span>Popularity of Music Records</a></div><div class="lev2"><a href="#Problem-1.1---Understanding-the-Data"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Problem 1.1 - Understanding the Data</a></div><div class="lev2"><a href="#Problem-1.2---Understanding-the-Data"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Problem 1.2 - Understanding the Data</a></div><div class="lev2"><a href="#Problem-1.4---Understanding-the-Data"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Problem 1.4 - Understanding the Data</a></div><div class="lev2"><a href="#Problem-1.5---Understanding-the-Data"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>Problem 1.5 - Understanding the Data</a></div><div class="lev2"><a href="#Problem-2.1---Creating-Our-Prediction-Model"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Problem 2.1 - Creating Our Prediction Model</a></div><div class="lev2"><a href="#Problem-2.2---Creating-our-Prediction-Model"><span class="toc-item-num">1.6&nbsp;&nbsp;</span>Problem 2.2 - Creating our Prediction Model</a></div>

Popularity of Music Records
===========================

The music industry has a well-developed market with a global annual revenue around $15 billion. 
The recording industry is highly competitive and is dominated by three big production companies 
which make up nearly 82% of the total annual album sales. 

Artists are at the core of the music industry and record labels provide them with the necessary resources
to sell their music on a large scale. A record label incurs numerous costs
(studio recording, marketing, distribution, and touring) in exchange for a percentage of the profits from album sales, singles and concert tickets.

Unfortunately, the success of an artist's release is highly uncertain: a single may be extremely popular, resulting in widespread radio play and digital downloads, while another single may turn out quite unpopular, and therefore unprofitable. 

Knowing the competitive nature of the recording industry, record labels face the fundamental decision problem of which musical releases to support to maximize their financial success. 

How can we use analytics to predict the popularity of a song? In this assignment, we challenge ourselves to predict whether a song will reach a spot in the Top 10 of the Billboard Hot 100 Chart.

Taking an analytics approach, we aim to use information about a song's properties to predict its popularity. The dataset songs.csv consists of all songs which made it to the Top 10 of the Billboard Hot 100 Chart from 1990-2010 plus a sample of additional songs that didn't make the Top 10. This data comes from three sources: Wikipedia, Billboard.com, and EchoNest.

The variables included in the dataset either describe the artist or the song, or they are associated with the following song attributes: time signature, loudness, key, pitch, tempo, and timbre.

Here's a detailed description of the variables:

- year = the year the song was released
- songtitle = the title of the song
- artistname = the name of the artist of the song
- songID and artistID = identifying variables for the song and artist
- timesignature and timesignature_confidence = a variable estimating the time signature of the song, and the confidence in the estimate
- loudness = a continuous variable indicating the average amplitude of the audio in decibels
- tempo and tempo_confidence = a variable indicating the estimated beats per minute of the song, and the confidence in the estimate
- key and key_confidence = a variable with twelve levels indicating the estimated key of the song (C, C#, . . ., B), and the confidence in the estimate
- energy = a variable that represents the overall acoustic energy of the song, using a mix of features such as loudness
- pitch = a continuous variable that indicates the pitch of the song
- timbre_0_min, timbre_0_max, timbre_1_min, timbre_1_max, . . . , timbre_11_min, and timbre_11_max = variables that indicate the - minimum/maximum values over all segments for each of the twelve values in the timbre vector (resulting in 24 continuous variables)
- Top10 = a binary variable indicating whether or not the song made it to the Top 10 of the Billboard Hot 100 Chart (1 if it was in the top 10, and 0 if it was not)

## Problem 1.1 - Understanding the Data

Use the read.csv function to load the dataset "songs.csv" into R.

How many observations (songs) are from the year 2010?

In [1]:
songs <- read.csv("songs.csv")
str(songs)

'data.frame':	7574 obs. of  39 variables:
 $ year                    : int  2010 2010 2010 2010 2010 2010 2010 2010 2010 2010 ...
 $ songtitle               : Factor w/ 7141 levels "'03 Bonnie & Clyde",..: 6204 5522 241 3098 47 607 254 4419 2887 6756 ...
 $ artistname              : Factor w/ 1032 levels "50 Cent","98 Degrees",..: 3 3 3 3 3 3 3 3 3 12 ...
 $ songID                  : Factor w/ 7549 levels "SOAACNI1315CD4AC42",..: 595 5439 5252 1716 3431 1020 1831 3964 6904 2473 ...
 $ artistID                : Factor w/ 1047 levels "AR00B1I1187FB433EB",..: 671 671 671 671 671 671 671 671 671 507 ...
 $ timesignature           : int  3 4 4 4 4 4 4 4 4 4 ...
 $ timesignature_confidence: num  0.853 1 1 1 0.788 1 0.968 0.861 0.622 0.938 ...
 $ loudness                : num  -4.26 -4.05 -3.57 -3.81 -4.71 ...
 $ tempo                   : num  91.5 140 160.5 97.5 140.1 ...
 $ tempo_confidence        : num  0.953 0.921 0.489 0.794 0.286 0.347 0.273 0.83 0.018 0.929 ...
 $ key                  

The songs dataset consists of 7574 observations.

In [3]:
table(songs$year)["2010"]

2010 
 373

From these 7574 observations, 373 are from the year 2010.

## Problem 1.2 - Understanding the Data

How many songs does the dataset include for which the artist name is "Michael Jackson"?

In [5]:
table(songs$artistname)["Michael Jackson"]

Michael Jackson 
             18

Problem 1.3 - Understanding the Data

Which of these songs by Michael Jackson made it to the Top 10? Select all that apply.

- Beat It
- You Rock My World
- Billie Jean
- You Are Not Alone

In [6]:
MJ <- subset(songs, artistname == "Michael Jackson")

In [22]:
MJ_songs <- MJ[c("songtitle", "Top10")]
subset(MJ_songs, MJ_songs$Top10 == 1)

songtitle Top10
4329 You Rock My World     1
6207 You Are Not Alone     1
6210    Black or White     1
6218 Remember the Time     1
6915     In The Closet     1

You Rock My World and You Are Not Alone made it to the top10.

## Problem 1.4 - Understanding the Data

The variable corresponding to the estimated time signature (timesignature) is discrete, meaning that 
it only takes integer values (0, 1, 2, 3, . . . ). 

What are the values of this variable that occur in our dataset?

In [23]:
table(songs$timesignature)


   0    1    3    4    5    7 
  10  143  503 6787  112   19 

Which timesignature value is the most frequent among songs in our dataset? --> 4

## Problem 1.5 - Understanding the Data

Out of all of the songs in our dataset, the song with the highest tempo is one of the following songs. Which one is it?

- Until The Day I Die
- Wanna Be Startin' Somethin'
- My Happy Ending
- You Make Me Wanna...

In [32]:
songs$songtitle[which.max(songs$tempo)]

[1] Wanna Be Startin' Somethin'
7141 Levels: '03 Bonnie & Clyde '69 ... Zumbi

## Problem 2.1 - Creating Our Prediction Model

We wish to predict whether or not a song will make it to the Top 10.

To do this, first use the subset function to split the data into a training set "SongsTrain" consisting of all the observations up to and including 2009 song releases, and a testing set "SongsTest", consisting of the 2010 song releases.

How many observations (songs) are in the training set?

In [33]:
SongsTrain <- subset(songs, songs$year <= 2009)
SongsTest <- subset(songs, songs$year > 2009)

In [34]:
nrow(SongsTrain)

[1] 7201

## Problem 2.2 - Creating our Prediction Model

In this problem, our outcome variable is "Top10" - we are trying to predict whether or not a song will make it to the Top 10 of the Billboard Hot 100 Chart. 

Since the outcome variable is binary, we will build a logistic regression model. We'll start by using all song attributes as our independent variables, which we'll call Model 1.

We will only use the variables in our dataset that describe the numerical attributes of the song in our logistic regression model. So we won't use the variables "year", "songtitle", "artistname", "songID" or "artistID".

Use the glm function to build a logistic regression model to predict Top10 using all of the other variables as the independent variables. You should use SongsTrain to build the model.

Looking at the summary of your model, what is the value of the Akaike Information Criterion (AIC)?

In [40]:
nonvars = c("year", "songtitle", "artistname", "songID", "artistID")

SongsTrain = SongsTrain[ , !(names(SongsTrain) %in% nonvars) ]
SongsTest = SongsTest[ , !(names(SongsTest) %in% nonvars) ]

In [41]:
SongsLog1 = glm(Top10 ~ ., data=SongsTrain, family=binomial)

In [42]:
summary(SongsLog1)


Call:
glm(formula = Top10 ~ ., family = binomial, data = SongsTrain)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.9220  -0.5399  -0.3459  -0.1845   3.0770  

Coefficients:
                           Estimate Std. Error z value Pr(>|z|)    
(Intercept)               1.470e+01  1.806e+00   8.138 4.03e-16 ***
timesignature             1.264e-01  8.674e-02   1.457 0.145050    
timesignature_confidence  7.450e-01  1.953e-01   3.815 0.000136 ***
loudness                  2.999e-01  2.917e-02  10.282  < 2e-16 ***
tempo                     3.634e-04  1.691e-03   0.215 0.829889    
tempo_confidence          4.732e-01  1.422e-01   3.329 0.000873 ***
key                       1.588e-02  1.039e-02   1.529 0.126349    
key_confidence            3.087e-01  1.412e-01   2.187 0.028760 *  
energy                   -1.502e+00  3.099e-01  -4.847 1.25e-06 ***
pitch                    -4.491e+01  6.835e+00  -6.570 5.02e-11 ***
timbre_0_min              2.316e-02  4.256e-03   5.44

Looking at the bottom of the summary(SongsLog1) output, we can see that the AIC value is 4827.2.